oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/IndexUpdate.java

```diff
@@ -243,7 +243,18 @@ private boolean shouldReindex(NodeBuilder definition, NodeState before,
         // See also OAK-7991.
         boolean result = !before.getChildNode(INDEX_DEFINITIONS_NAME).hasChildNode(name)
                 && !hasAnyHiddenNodes(definition);
-        if (result) {
+        // See OAK-9449
+        // In case of elasticsearch, indexed data is stored remotely and not under hidden nodes, so
+        // in case of OutOfBand indexing during content import, there is no hidden node created for elastic (not even :status)
+        // So, we log a warn and return false to avoid unnecessary reindexing. The warning is only if the someone added the new index node and forgot to add
+        // the reindex flag, in case OutOfBand Indexing has been performed, warning can be ignored.
+        // Also, in case the new elastic node has been added with reindex = true , this method would have already returned true
+        if (result && TYPE_ELASTICSEARCH.equals((type.getValue(Type.STRING)))) {
+            log.warn("Found a new elastic index node [{}]. Please set the reindex falg = true to initiate reindexing." +
```

Comment:
```suggestion
            log.warn("Found a new elastic index node [{}]. Please set the reindex flag = true to initiate reindexing." +
```

Reply:
> Would it be possible to set an internal property in the ElasticIndexDefinition constructor and then check here if this 
> property` exists?

@fabriziofortino  - yes could be possible, so we can possible add some property like isElasticIndex = true, but IMO, that just makes things more confusing (because to everywhere we are using the TYPE property to identify what type of index this is ) and here we wanted to handle this differently for elastic type index. And even if we add the new property, we would still be having this check specifically for elastic. (Or maybe you were trying to suggest something different ?)

Reply:
I was thinking about some more generic property that does not mention the type at all. Not sure what the best name could be: `:reindexEnabled`, `:reindexable`?
If we will add another remote index type we don't need to change `IndexUpdate` to include that.

Reply:
what about :autoReindexNewNode ?

Reply:
it sounds good to me. I would put that in a public constant in `IndexUpdate` and access it from `ElasticIndexDefintion`.

Reply:
@fabriziofortino  - one problem that I see with this approach is that I don't think we use ElasticIndexDefinition while actually storing the index def in the repo - it's only used at the time of querying (I was actually confusing it earlier with ElasticIndexDefinitionBuilder, which unfortunately is also not used while creating the elastic index defs - and as of now we won't be able to use it in GraniteContent, because of lucene/elastic dep conflict.)

The IndexUpdate class is in oak-core, and from the first looks of it I don't think it's possible to get a ElasticIndexDefinition instance there, I will see if it's possible somehow.

Reply:
I don't think we should access `ElasticIndexDefinition` from `IndexUpdate`. This won't be correct and even possible. I was just thinking to access a property from the `NodeState` that was previously set in the `oak-elastic` module. I don't know if this would be possible either since I am not familiar with the application logic in `IndexUpdate`. Probably looking at a stack trace would tell us if this is possible or not.

Reply:
@fabriziofortino  - ok, I get it now. But the problem here is that on skyline, the indexing job will create indexing definitions from a json, and customers can also create definitions via content packages. And then the job applies it to the repo via the IndexDefinitionUpdater class which basically serializes the json to create the index def. We could possibly add a rule there to add this hidden property for elastic type, but then it would be the same problem again. 

And I don't want to make this property public, because it's just making creation of the index definitions more complicated for the customers.

So I think, let's stick with the elastic type check in IndexUpdate for now.

Reply:
👍